### PR TITLE
Add XGBoostGqVariantFilter, a tool to recalibrate GQ

### PR DESCRIPTION
**Pull request by TedBrookings**
* Added MinGqVariantFilterBase
* * loads VCF, pedigree, UCSC genome tract, and truth data
* * calculates variant overlap with genome tracts
* * forms matrices, tensors, and other helping data for machine learning
* * provides for TRAIN and FILTER modes
* * provides functions for calculating loss given assigned min GQ values
* * computes best estimate of truth data used for training xgboost model
* Added XGBoostMinGqVariantFilter
* * calculates new GQ based on gradient boosting
* Added PropertiesTable for loading VCF properties into tensors
* Added TractOverlapDetector for computing overlap properties with
  UCSC genome tracts

Training loss is based on weighted combination of heredity and truth
data, broken down by variant category.

### CODE REVIEW

**Comment by TedBrookings** [comment]
There's a lot of stuff that I know is wrong here:

1. This is based on a master that's super out of date (I don't want to rebase at this juncture, because I'd need to stop and verify that behavior didn't change due to something else changing in GATK)
2. No unit tests. Up to this point, the basic structure has been changing a lot. It should be pretty well settled now though.
3. Probably the main classes should be renamed to indicated that they are recalibrating GQ, not just filtering.
4. I should probably put in a soft-filter option (just recalibrate GQ, don't set GT to no-call)
5. Probably the output should be called something other than GQ. Phred-scaling is a bad match to probabilities near 50%, but people expect GQ to be Phred-scaled.
